### PR TITLE
Feature/root alias for plotting

### PR DESCRIPTION
**Pull request by BW177**
This PR implements the option of aliasing module names for the sake of shortening plot labels.
It takes advantage of networkx.draw_networkx "labels" argument, i.e. it takes supplied aliases into account when generating the labels. 
Since "labels" is a detail of networkx, the implementation was kept within networkxgraph.py. This avoids higher layers (e.g. evaluable_architecture.py) depending on details of networkx. A user would plot architecture by calling `EvaluableArchitectureGraph`'s `visualize()` method, thus tests were written against this layer. 

I ran into problems where poetry does only install python 3.10 which lead to runtime errors when trying to run the pre-commit hooks and to nox skipping tests with python 3.8 and 3.9. If you have an idea how to fix that please let me know.

### CODE REVIEW

File: src/pytestarch/eval_structure_impl/networkxgraph.py [hunk 1]
```diff
@@ -198,14 +198,77 @@ def parent_child_relationship(
...
 
+        if "aliases" in kwargs:
+            aliases = kwargs.pop("aliases")
+            labels = self._create_plot_labels_with_alias(aliases)
+            kwargs["labels"] = labels
+
         draw_networkx(self._graph, **kwargs)
 
+    def _create_plot_labels_with_alias(self, aliases: dict[str, str]) -> dict[str, str]:
+        module_names: list[str] = self._graph.nodes
```
Comment (zyskarch) [inline src/pytestarch/eval_structure_impl/networkxgraph.py:231]: type highlighting: if we wrap a list(...) around the nodes, the warning highlight is gone

File: tests/eval_structure/test_evaluable_graph.py [hunk 2]
```diff
@@ -840,3 +841,75 @@ def test_not_other_to_module_than_between_submodule_modules(
...
+    module_1_alias = "1"
+    aliases = {MODULE_1: module_1_alias}
+    fig, ax = subplots()
+
+    evaluable.visualize(aliases=aliases, ax=ax)
+
+    labels_in_plot = [t.get_text() for t in ax.texts]
+    assert module_1_alias in labels_in_plot
+    assert MODULE_1 not in labels_in_plot
+    assert SUB_MODULE_OF_1 not in labels_in_plot
```
Comment (zyskarch) [inline tests/eval_structure/test_evaluable_graph.py:869]: Maybe also assert the aliases name of the submodule?

**Review by zyskarch — APPROVED** [review]

**Comment by zyskarch** [comment]
Closes #32

File: src/pytestarch/eval_structure_impl/networkxgraph.py (hunk 1 above)
Comment (BW177) [reply src/pytestarch/eval_structure_impl/networkxgraph.py:231]: done

**Review by zyskarch — APPROVED** [review]

**Pull request merged**